### ISSUE TITLE
FP: The parameter is overridden immediately (original value is lost)

### ISSUE THREAD

**Issue by florianajir** [kalessil/phpinspectionsea#1325]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended)  |
| Language level | PHP 7.1                         |

Current behaviour
--
Throw a false positive error

Expected behaviour
--
Nothing

Environment details
---
How to reproduce:

```php
$validationErrors = [];
if (count($validationErrors)) {}
```

**Comment by orklah** [comment]
Hi,

I can't reproduce the issue with your example.

Could you give further details or a larger example who could show the issue?

**Comment by kalessil** [comment]
Thank you for reporting @florianajir, I also could not reproduce. Would you please share an extended code fragment, please?

**Comment by florianajir** [comment]
It seems to be an ide init bug or I don't know what but the error did not thrown today.
However that was not the first time I seen this false positive error.

But you can close the issue because I'm not able to reproduce anymore.

**Comment by kalessil** [comment]
Ok, it would be great to get a reproducible sample if you run into the issue again. meanwhile closing,

**Comment by florianajir** [comment]
It happens again.. :

![image](https://user-images.githubusercontent.com/3715903/58106335-44e50680-7be8-11e9-8db2-bfce2a343b72.png)

**Comment by florianajir** [comment]
It happens again 

![image](https://user-images.githubusercontent.com/3715903/58106405-59290380-7be8-11e9-9d3a-c1fcb6e4a88d.png)

![image](https://user-images.githubusercontent.com/3715903/58106476-7eb60d00-7be8-11e9-8094-3d23af29a012.png)

**Comment by kalessil** [comment]
I can see now, looks like a correctly reported case: `$validationErrors` is overridden and not being used between the declaration and the override. In this case, renaming the local variable is recommended.

**Comment by florianajir** [comment]
where do you see an override ?

**Comment by orklah** [comment]
$validationErrors is a parameter of your method. You change its value without using it once. That's why Kalessil recommanded to use a different name for your variable, to avoid confusion between the value passed in parameter and the changed value.

**Comment by florianajir** [comment]
I don't understand the point "You change its value without using it once"

**Comment by florianajir** [comment]
Ok I didn't notice the parameter of the method, my bad. Thank you I close the issue.

**Comment by kalessil** [comment]
@florianajir, just a second =) I'm thinking if warning message needs to be adjusted, perhaps you can propose how it can be improved?

**Comment by florianajir** [comment]
Indeed reading the message i supposed the problem come after the declaration.
Maybe something like "This variable name has already been declared previously without being used"

**Comment by kalessil** [comment]
@florianajir: updated!
